docs/index.rst

```diff
@@ -1,23 +1,108 @@
-.. pluggy documentation master file, created by
-   sphinx-quickstart on Mon Nov 14 11:08:31 2016.
-   You can adapt this file completely to your liking, but it should at least
-   contain the root `toctree` directive.
+``pluggy``
+==========
 
-Welcome to pluggy's documentation!
-==================================
+The ``pytest`` plugin system
+----------------------------
+``pluggy`` is the crystallized core of `plugin management and hook
+calling`_ for `pytest`_.
 
-Contents:
+In fact, ``pytest`` is itself composed as a set of ``pluggy`` plugins
+which are invoked in sequence according to a well defined set of protocols.
+Some 150+ plugins use ``pluggy`` to extend and customize ``pytest``'s default behaviour.
 
+In essence, ``pluggy`` enables function `hooking`_ so you can build "pluggable" systems.
+
+How's it work?
+--------------
+A `plugin` is a `namespace`_ which defines hook functions.
+
+``pluggy`` manages *plugins* by relying on:
+
+- a hook *specification* - defines a call signature
+- a set of hook *implementations* - aka `callbacks`_
+- the hook *caller* - a call loop which collects results
+
+where for each registered hook *specification*, a hook *call* will invoke up to ``N``
+registered hook *implementations*.
+
+``pluggy`` accomplishes all this by implementing a `request-response pattern`_ using *function*
+subscriptions and can be thought of and used as a rudimentary busless `publish-subscribe`_
+event system.
+
+``pluggy``'s approach is meant to let a designer think carefuly about which objects are
+explicitly needed by an extension writer. This is in contrast to subclass-based extension
+systems which may expose unecessary state and behaviour or encourage `tight coupling`_
+in overlying frameworks.
+
+
+A first example
+---------------
+
+.. literalinclude:: examples/firstexample.py
+
+Running this directly gets us::
+
+    $ python docs/examples/example1.py
+
+    inside Plugin_2.myhook()
+    inside Plugin_1.myhook()
+    [-1, 3]
+
+For more details and advanced usage see our
+
+User Guide
+----------
 .. toctree::
-   :maxdepth: 2
+   :maxdepth: 1
 
+   define
+   manage
+   calling
+   tracing
    api_reference
 
+.. tracing
+
+Development
+-----------
+Great care must taken when hacking on ``pluggy`` since many mature
+projects rely on it. Until there is a more automated CI process in
```

Comment:
I think this part about "automated CI process" can be removed since we have AppVeyor and Travis automated testing.